packages/storage-plugin/src/storage.plugin.ts

```diff
@@ -57,41 +55,42 @@ export class NgxsStoragePlugin implements NgxsPlugin {
         }
 
         const isMaster = key === DEFAULT_STATE_KEY;
-        let val: any = this._engine.getItem(key!);
-
-        if (val !== 'undefined' && val != null) {
-          try {
-            const newVal = this._options.deserialize!(val);
-            val = this._options.afterDeserialize!(newVal, key);
-          } catch (e) {
-            // Caretaker note: we have still left the `typeof` condition in order to avoid
-            // creating a breaking change for projects that still use the View Engine.
-            if (typeof ngDevMode === 'undefined' || ngDevMode) {
-              console.error(
-                `Error ocurred while deserializing the ${key} store value, falling back to empty object, the value obtained from the store: `,
-                val
-              );
-            }
-            val = {};
+        let val: any;
+        try {
+          val = this._engine.getItem(key!);
+          if (val === 'undefined' || val == null) {
+            continue;
```

Comment:
This `continue;` here will cause an `undefined` or `null` value to skip running through the migrations, which is a change from previous behaviour, and potentially a breaking change for some production use cases.

Reply:
I starred at this for a while and started looking at alternatives.
Once I reverted the code I realised this is the previous behaviour. I just reversed the condition like a guard, easier to read.
`if (val !== 'undefined' && val != null) {` <--
@markwhitfeld Are you thinking about something else?

Reply:
Hi Marcel. Apologies for missing your response again.
When you inverted the condition and added the `continue;` code, the code execution will skip all of the rest of the code in the `for (const key of keys) {` loop and move to the next iteration. This will therefore skip the `if (this._options.migrations) {` section of code, which was not the case before.
I haven't verified what this subsequent code would do with a value of `null` or `'undefined'` (which could potentially be an edge case that has not been tested and exhibits some reasonable behavior, or even a bug). But this change is nevertheless an unintended change to program flow.

Do you think that there is a different way to make your changes and preserve the program flow to avoid this side effect?